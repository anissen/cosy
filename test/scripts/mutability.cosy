
mut a = 5
print a
a = 10
print a

let b = 3
print b
b = 7
print b

fn f(arg) {
    arg = 5
    print arg
}

mut x = 0
print x
f(x)
print x
