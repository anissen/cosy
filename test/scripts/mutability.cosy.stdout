■ test/scripts/mutability.cosy, line 9:
 7 | let b = 3
 8 | print b
 9 | b = 7
     ^ Cannot reassign non-mutable variable.
10 | print b
11 | 

■ test/scripts/mutability.cosy, line 13:
11 | 
12 | fn f(arg) {
13 |     arg = 5
         ^^^ Cannot reassign non-mutable variable.
14 |     print arg
15 | }

